### PR TITLE
Deprecate ScimHelper class

### PR DESCRIPTION
**Pull request by wallner**
Mark dubious and untested class `ScimHelper` as `@Deprecated` and move
its only provided method where it belongs to the `User` class. Also adds
tests for the method.

### CODE REVIEW

**Comment by dacrome** [comment]
Thanks. Please make sure to change it in the [addon-self-administration](https://github.com/osiam/addon-self-administration/search?utf8=%E2%9C%93&q=getPrimaryOrFirstEmail) and [addon-administration](https://github.com/osiam/addon-administration/search?utf8=%E2%9C%93&q=getPrimaryOrFirstEmail) too.

**Comment by tkrille** [comment]
@dacrome You want to assign yourself?

**Comment by tkrille** [comment]
Please double check that the new "property" does not end up in the JSON.

File: src/main/java/org/osiam/resources/scim/User.java [hunk 1]
```diff
@@ -279,6 +280,27 @@ public String getPassword() {
...
 
     /**
+     * try to extract an email from the User.
+     * If the User has a primary email address this email will be returned.
+     * If not the first email address found will be returned.
+     * If no Email has been found email.isPresent() == false
+     *
+     * @return an email if found
+     */
+    public Optional<Email> getPrimaryOrFirstEmail() {
```
Comment (dacrome) [inline src/main/java/org/osiam/resources/scim/User.java]: Thanks @tkrille for the hint. The getter needs to be annotated with `@JsonIgnore` to be ignored in the json representation.

**Comment by wallner** [comment]
Thank you for catching that @tkrille and @dacrome, I should not do PRs when I'm tired. I will take care of it.

File: src/test/groovy/org/osiam/resources/scim/UserSpec.groovy [hunk 2]
```diff
@@ -301,10 +300,52 @@ class UserSpec extends Specification {
         newUser = new User.Builder(newUserName, oldUser).build()
 
         then:
-        newUser.isActive() == true
+        newUser.isActive()
```
Comment (dacrome) [inline src/test/groovy/org/osiam/resources/scim/UserSpec.groovy]: I think explicit assertions on boolean values are easier to read.

File: src/test/groovy/org/osiam/resources/scim/UserSpec.groovy [hunk 3]
```diff
@@ -444,7 +485,7 @@ class UserSpec extends Specification {
         user.isActive() == newUser.isActive()
 
         newUser.getExtension('urn:scim:schemas:extension:test:1.0:User').getFieldAsString('keyString') == 'example'
-        newUser.getExtension('urn:scim:schemas:extension:test:1.0:User').getFieldAsBoolean('keyBoolean') == true
+        newUser.getExtension('urn:scim:schemas:extension:test:1.0:User').getFieldAsBoolean('keyBoolean')
```
Comment (dacrome) [inline src/test/groovy/org/osiam/resources/scim/UserSpec.groovy]: Same here.

File: src/test/groovy/org/osiam/resources/scim/UserSpec.groovy (hunk 2 above)
Comment (wallner) [inline src/test/groovy/org/osiam/resources/scim/UserSpec.groovy]: I disagree. Violently, I don't think we should get started down that slippery slope. I usually replace explicit comparisons with "true" or "false". What's hard to read in the line above? We might need a tie breaker here, @tkrille?

File: src/test/groovy/org/osiam/resources/scim/UserSpec.groovy (hunk 3 above)
Comment (wallner) [inline src/test/groovy/org/osiam/resources/scim/UserSpec.groovy]: See above.

File: src/test/groovy/org/osiam/resources/scim/UserSpec.groovy (hunk 2 above)
Comment (tkrille) [inline src/test/groovy/org/osiam/resources/scim/UserSpec.groovy]: I personally like the explicit style with operator and operand more. But that's just my opinion and taste. As of this, for me there's nothing to discuss. If @wallner likes the implicit style more, then it's ok and we shouldn't force nobody to adhere to someone else's personal style. What might matter is consistency. But that might also be bikeshedding :)

Comment (dacrome) [inline src/test/groovy/org/osiam/resources/scim/UserSpec.groovy]: Ok, I would agree if @wallner did this in his newely created test code, but he removed it from already written test code. We should just use one style and I prefer the explicit.

Comment (tkrille) [inline src/test/groovy/org/osiam/resources/scim/UserSpec.groovy]: We settled this in private with the following outcome: the implicit style favored by @wallner is the recommended one from now on.

**Comment by dacrome** [comment]
Thank you (also for to be patient because of the out-of-scope discussion about the comparison of boolean values in unit tests ;) )

Please do not forget to update the addons. I've already [commented this above](https://github.com/osiam/scim-schema/pull/157#issuecomment-143478039)

**Comment by wallner** [comment]
I'm going to do the addons later tonight.

**Pull request merged**